src/asset.ts

```diff
@@ -21,14 +21,25 @@ export default class Asset {
     ethAddress: string,
     contractAddress?: string,
   ) {
-    const data = { appId, timestamp: Date.now() };
+    const timestamp = Date.now();
+    const query = { appId };
+    const trailingUrl = contractAddress
+      ? `nft/${chainId}/${ethAddress}/${contractAddress}`
+      : `nft/${chainId}/${ethAddress}`;
+    const data = {
```

Comment:
would it be useful to add a util function like `buildQueryData(method: string, path: string, timestamp: ...)`?

Reply:
Done! Add `buildData` util function!